crates/joystick/src/common.rs

```diff
@@ -0,0 +1,75 @@
+use evdev::KeyCode;
+use std::time::Duration;
+
+pub static TAP_DURATION: u64 = 50;
+
+pub enum KeyAction {
+    Press,
+    Release,
+}
+
+pub trait JoystickInterface {
+    fn press_a(&mut self);
+    fn press_b(&mut self);
+    fn press_x(&mut self);
+    fn press_y(&mut self);
+    fn press_lt(&mut self);
+    fn press_rt(&mut self);
+    fn press_lt2(&mut self);
+    fn press_rt2(&mut self);
+    fn press_select(&mut self);
+    fn press_start(&mut self);
+    fn press_dpad_up(&mut self);
+    fn press_dpad_down(&mut self);
+    fn press_dpad_left(&mut self);
+    fn press_dpad_right(&mut self);
+
+    fn release_a(&mut self);
+    fn release_b(&mut self);
+    fn release_x(&mut self);
+    fn release_y(&mut self);
+    fn release_lt(&mut self);
+    fn release_rt(&mut self);
+    fn release_lt2(&mut self);
+    fn release_rt2(&mut self);
+    fn release_select(&mut self);
+    fn release_start(&mut self);
+    fn release_dpad_up(&mut self);
+    fn release_dpad_down(&mut self);
+    fn release_dpad_left(&mut self);
+    fn release_dpad_right(&mut self);
+
+    fn tap_a(&mut self);
+    fn tap_b(&mut self);
+    fn tap_x(&mut self);
+    fn tap_y(&mut self);
+    fn tap_lt(&mut self);
+    fn tap_rt(&mut self);
+    fn tap_lt2(&mut self);
+    fn tap_rt2(&mut self);
+    fn tap_select(&mut self);
+    fn tap_start(&mut self);
+    fn tap_dpad_up(&mut self);
+    fn tap_dpad_down(&mut self);
+    fn tap_dpad_left(&mut self);
+    fn tap_dpad_right(&mut self);
+    fn release_all(&mut self);
+    fn run(&mut self);
+
+    #[cfg(target_os = "linux")]
```

Comment:
I don't think these should be here. They are all hidden methods used by the implementations, so they should be put in those files (in a regular `impl` block, not related to the trait).

Reply:
agreed